docs/guides/seccomp.md

```diff
@@ -0,0 +1,281 @@
+to start the trace before the pod is up, then bring up the pod and generate
+traffic.
+
+We can stop the current trace using the `stop` operation, and delete the
+current pod, so that we can start a fresh new trace.
+
+```
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=stop
+trace.gadget.kinvolk.io/seccomp annotated
+$ kubectl delete -f docs/examples/seccomp/unconfined.yaml
+pod "hello-python" deleted
+```
+
+Now we can `start` a new trace, and then create the pod again.
+
+```
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=start
+trace.gadget.kinvolk.io/seccomp annotated
+$ kubectl apply -f docs/examples/seccomp/unconfined.yaml
+pod/hello-python created
+```
+
+Once the pod is up, we can once again generate some traffic, like before.
+
+```
+$ kubectl get svc hello-python-service -n seccomp-demo
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+hello-python-service   LoadBalancer   10.111.82.101   <pending>     6000:31863/TCP   72s
+$ curl 10.111.82.101:6000
+Hello World!
+```
+
+And now generate the policy again.
+```
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=generate
+trace.gadget.kinvolk.io/seccomp annotated
+```
+
+This time, the Status field will contain a lot more syscalls, as a lot of
+operations need to take place to bring up the pod. 
+
+### Integration with Kubernetes Security Profiles Operator
+
+We can use the output stored in the trace to create the seccomp policy for
+our pod. But instead of copying it manually, we can also use the
+integration with the [Kubernetes Security Profiles
+Operator](https://github.com/kubernetes-sigs/security-profiles-operator).
+
+When this operator is
+[installed](https://github.com/kubernetes-sigs/security-profiles-operator/blob/master/installation-usage.md#install-operator)
+in the cluster, the seccomp gadget can generate `seccompprofile` resources
+that can be used directly by our pods.
+
+To do this, we need to change the `outputMode` field to `ExternalResource` and
+the `output` field to the namespace and name of the desired resource. You
+can edit the trace file and re-apply it. Our example directory already
+includes a modified version of the trace in `trace-external.yaml`:
+
+```
+apiVersion: gadget.kinvolk.io/v1alpha1
+kind: Trace
+metadata:
+  name: seccomp
+  namespace: gadget
+spec:
+  node: demo-node
+  gadget: seccomp
+  filter:
+    namespace: seccomp-demo
+    podname: hello-python
+  runMode: Manual
+  outputMode: ExternalResource
+  output: seccomp-demo/hello-profile
+```
+
+**Remember that you need to set the right node name**.
+
+We can apply this modified file, which will change the existing resource
+(keeping the already gathered data) and then call `generate` again to get
+the profile generated:
+
+```
+$ kubectl apply -f docs/examples/seccomp/trace-external.yaml
+trace.gadget.kinvolk.io/seccomp configured
```

Comment:
I wonder if that's something we should advertise...

In my PR "[seccomp: generate SeccompProfile for terminated pods](https://github.com/kinvolk/inspektor-gadget/pull/272)", I generate Seccomp Profiles directly upon pod termination, and no longer only after annotating with "generate". In that case, the "OutputMode" setting is read when reading the "start" operation, and changing it afterwards will not change the setting for those terminated pods.

Reply:
Right, I didn't expect this to work myself, and discovered it just by chance when preparing the demo. It makes sense not to advertise it if we don't want to stick to this behavior.

Reply:
I've redone this part without mentioning that calling generate directly would work. This means re-doing all the previous steps, which is a bit annoying, but I guess it's fine to just re-iterate what's going on.